<a href="https://colab.research.google.com/github/japndisal/flyrank-starter-repo/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/japndisal/flyrank-starter-repo/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

**My lane: Refresh / Content Opportunity Scoring**

I'm choosing this lane because it starts with understanding which search and content signals are actually associated with visibility and clicks, rather than jumping straight to a ranking model. Compared with the other lanes, it sits where analytics and ML meet: I can begin with simple, explainable evidence (EDA, correlations, grouped comparisons) and then, if the signals hold up, build toward a scoring model that helps a content team decide which pages to review first. It also fits the data I have, since many pages show both real demand (impressions) and signs of decline.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Decision this improves**: which of 40+ available signals a content/SEO team should actually pay attention to when triaging pages, instead of treating every metric as equally important.

**Who acts on it**: a content strategist or SEO lead deciding where to spend limited review time.

**Action taken**: prioritizing monitoring/review around the 2–3 signals shown to genuinely associate with performance (e.g. position, freshness), instead of spreading effort thin.

**Cost of a wrong call**: flagging a signal as meaningful when it isn't wastes analyst time chasing a pattern that doesn't hold (e.g. re-optimizing for word count if it barely correlates with anything). Missing a real signal means under-investing in the one lever that actually moves clicks. Since this is EDA/decision-support, not a live experiment, the cost is wasted time and misplaced priorities — not broken production traffic.

In [5]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [6]:
# This cell is for CODE (numbers, a query, a check).
import pandas as pd

path = "https://raw.githubusercontent.com/japndisal/flyrank-starter-repo/main/data/raw/content_refresh_anonymized.csv"
df = pd.read_csv(path)
print("Shape:", df.shape)

# Trend breakdown
trend_counts = df["trend_direction"].value_counts(dropna=False)
print("\nTrend direction counts:")
print(trend_counts)

# Simple flags
down = df["trend_direction"] == "down"
high_imp = df["impressions_90d"] >= 500
high_sess = df["sessions_90d"] >= 30
both = down & high_imp

print(f"\nPages trending down:            {down.sum():,} ({down.mean():.1%})")
print(f"Pages with >=500 impressions:   {high_imp.sum():,} ({high_imp.mean():.1%})")
print(f"Pages with >=30 sessions:       {high_sess.sum():,} ({high_sess.mean():.1%})")
print(f"Down AND >=500 impressions:     {both.sum():,} ({both.mean():.1%})")
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


Shape: (30000, 44)

Trend direction counts:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Pages trending down:            16,262 (54.2%)
Pages with >=500 impressions:   16,726 (55.8%)
Pages with >=30 sessions:       7,114 (23.7%)
Down AND >=500 impressions:     9,961 (33.2%)


**Quick look at the starter data**

The starter file has 30,000 pages. Three numbers make this lane look worth the next 7 weeks:

1. **54.2% of pages are trending down** (by the simple `trend_direction == "down"` proxy), so there is a large pool of weaker pages that could go into a refresh queue.
2. **16,726 pages (about 55.8%) have at least 500 impressions in 90 days**, and 7,114 (about 23.7%) have at least 30 sessions. Many pages have real demand, so small improvements could matter.
3. **9,961 pages are both trending down and have at least 500 impressions (33.2% of the site).** These are the pages where decline and demand overlap, so they are the first candidates for review.

Together these suggest there are plenty of pages with both demand and signs of weakness, so deciding which to review first is a useful problem. These are observed counts from one dataset, not proof that a refresh would fix anything.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

**What I can claim**
This analysis is observational. The signals can described which are associate with visibility, clicks, or decline in this dataset, and how strongly, using correlation and grouped comparisons. Findings are directional, meant for decision-support: they suggest where to look first, not a guarantee.

**What I can't claim**
No signal here causes a ranking or click change. No sign of A/B test or intervention behind it. I'm not predicting or reverse-engineering any search engine's algorithm, I'm describing patterns in one anonymized dataset. A correlation like position-vs-CTR won't necessarily hold at the same strength for every client or content type, and will be flagged a whenever I report a number.

In [7]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.